File: snippets/python/recommend_batch_points.py
from qdrant_client import QdrantClient, models

client = QdrantClient(url="http://localhost:6333")

filter_ = models.Filter(must=[models.FieldCondition(key="city", match=models.MatchValue(value="London"))])

recommend_queries = [
    models.RecommendRequest(
        positive=[100, 231], negative=[718], filter=filter_, limit=3
    ),
    models.RecommendRequest(positive=[200, 67], negative=[300], limit=3),
]

client.recommend_batch(collection_name="{collection_name}", requests=recommend_queries)

File: snippets/go/recommend_batch_points.go
package client

import (
	"context"

	"github.com/qdrant/go-client/qdrant"
)

func recommendBatch() {
	client, err := qdrant.NewClient(&qdrant.Config{
		Host: "localhost",
		Port: 6334,
	})
	if err != nil {
		panic(err)
	}

	client.QueryBatch(context.Background(), &qdrant.QueryBatchPoints{
		CollectionName: "{collection_name}",
		QueryPoints: []*qdrant.QueryPoints{
			{
				CollectionName: "{collection_name}",
				Query: qdrant.NewQueryRecommend(&qdrant.RecommendInput{
					Positive: []*qdrant.VectorInput{
						qdrant.NewVectorInputID(qdrant.NewIDNum(100)),
						qdrant.NewVectorInputID(qdrant.NewIDNum(231)),
					},
					Negative: []*qdrant.VectorInput{
						qdrant.NewVectorInputID(qdrant.NewIDNum(718)),
					},
				}),
			},
			{
				CollectionName: "{collection_name}",
				Query: qdrant.NewQueryRecommend(&qdrant.RecommendInput{
					Positive: []*qdrant.VectorInput{
						qdrant.NewVectorInputID(qdrant.NewIDNum(200)),
						qdrant.NewVectorInputID(qdrant.NewIDNum(67)),
					},
					Negative: []*qdrant.VectorInput{
						qdrant.NewVectorInputID(qdrant.NewIDNum(300)),
					},
				}),
			},
		},
	})
}
